loopd/daemon.go

```diff
@@ -327,7 +314,26 @@ func (d *Daemon) stop() error {
 		d.clientCleanup()
 	}
 
-	return nil
+	// The servers should already have stopped gracefully. If not, we
+	// force them by closing its underlying listeners.
+	if d.grpcListener != nil {
+		_ = d.grpcListener.Close()
```

Comment:
log these errors?

Reply:
These would only return `use of closed network connection` because the servers themselves already closed the listeners. That's why I removed these lines completely.